firmware/application/io_file.cpp

```diff
@@ -19,20 +19,76 @@
  * Boston, MA 02110-1301, USA.
  */
 
+#include "string.h"
 #include "io_file.hpp"
 
+namespace fs = std::filesystem;
+static const fs::path c8_ext = u".C8";
+
+// convert c16 buffer to c8 buffer
+// (output buffer size is bytes/2)
+void FileWriter::c16_to_c8(const void* buffer, File::Size bytes) {
+    int16_t* src = (int16_t*)buffer;
+    uint16_t* dest = (uint16_t*)buffer;
+
+    for (File::Size i = 0; i < bytes / 4; i++) {
+        uint8_t re = *src++ / 256;
```

Comment:
Nit: Shifts instead of divide? Also consider some parens on `*(src++)`  or something to make the precedence clearer.

Reply:
I was hoping that the compiler would optimize this but I can try shift.  My concern was whether the compiler would do a right shift of a negative number correctly (filling with 1's from the left), which seems to be "implementation defined".

Reply:
Why work in uint if you’re expecting negative values?

Reply:
Converting to Uint because of the shift & addition a few lines below which may be problematic with signed numbers (I'm only combining the real & imaginary so I can do a single memory write for each complex number; the code definitely looks neater if I just write individual bytes though).

I experimented with shift in place of the divide by 256, and it results in a little spike at the center frequency, I guess because of the signed numbers, or perhaps it's just showing the real spike.  I will look at that further.

Reply:
I cleaned this part up; now using complex types and shifts as in proc_replay.